packages/ap-audit/alliance_platform/audit/settings.py

```diff
@@ -0,0 +1,71 @@
+from typing import TypedDict
+
+from alliance_platform.base_settings import AlliancePlatformSettingsBase
+from django.db.models import Value
+from django.db.models.expressions import Func
+from django.db.models.functions import Concat
+
+
+class AlliancePlatformAuditSettingsType(TypedDict, total=False):
+    """Settings for the Audit package of the Alliance Platform
+
+    These can be set in the Django settings file under the ``ALLIANCE_PLATFORM`` key:
+
+    .. code-block:: python
+
+        ALLIANCE_PLATFORM = {
+            "AUDIT": {
+                # settings go here
+            }
+        }
+    """
+
+    #: The name of the action used to evaluate permissions for auditing a model. Defaults to 'audit'.
+    LIST_PERM_ACTION: str | None
+    #: The name of the permission used to evaluate whether auditing of any type can be performed. If it resolves to ``False``,
+    #: this will override individual audit permissions determined by ``LIST_PERM_ACTION``. Defaults to
+    #: ``"alliance_platform.audit.can_audit"``
+    GLOBAL_AUDIT_PERMISSION_NAME: str | None
+    #: The SQL function used to generate the display name for users in audit logs. Defaults to showing first name and last name
+    USERNAME_FORMAT: Func | None
+    #: The format string to use when displaying timestamps for audit logs. Defaults to ``"%Y-%m-%d %H:%M:%S"``
+    DATETIME_FORMAT: str | None
```

Comment:
hmm why do we need to format it here? don't we just return the date in the API, so couldn't it be a standard date string? (maybe i'm wrong about that, haven't checked the UI code)

Reply:
I'm not entirely sure how useful this is. It carried over from `common_audit`. The date is converted to a string for serialisation specifically.